tests/testthat/test-board-rsconnect.R

```diff
@@ -16,12 +16,26 @@ test_that("User-supplied html files can overwrite the default", {
   expect_equal(readLines(file.path(dir, "index.html")), "new_file")
 })
 
-test_that("Mismatched protocols generate correct URL", {
+test_that("Mismatched protocols and ports generate correct URL", {
   path <- rsconnect_remote_path_from_url(list(server = "https://foo.com/rsc"), "http://foo.com/rsc/foo/bar")
   expect_equal(path, "/foo/bar")
 
   path <- rsconnect_remote_path_from_url(list(server = "http://foo.com/rsc"), "https://foo.com/rsc/foo/bar")
   expect_equal(path, "/foo/bar")
+
+  path <- rsconnect_remote_path_from_url(list(server = "https://foo.com:443/rsc"), "https://foo.com/rsc/foo/bar")
+  expect_equal(path, "/foo/bar")
+
+  path <- rsconnect_remote_path_from_url(list(server = "https://foo.com/rsc"), "https://foo.com:443/rsc/foo/bar")
+  expect_equal(path, "/foo/bar")
+
+
+  path <- rsconnect_remote_path_from_url(list(server = "https://foo.com:8443/rsc"), "https://foo.com:8443/rsc/foo/bar")
+  expect_equal(path, "/foo/bar")
+
+
+  path <- rsconnect_remote_path_from_url(list(server = "https://foo.com:8443/rsc"), "http://foo.com:8443/rsc/foo/bar")
+  expect_equal(path, "/foo/bar")
 })
```

Comment:
Suggesting a new test case based on #379 that seems to be fixed by some changes in this PR.  (because of trailing / added after server now)

it is not about protocol and port, so maybe in a new `test_that` call ? 

```r
path <- rsconnect_remote_path_from_url(list(server = "https://foo.com"), "http://foo.com/bar/foo-com_baz")
expect_equal(path, "/bar/foo-com_baz")
```

With current version (not this PR) 
```r
pins:::rsconnect_remote_path_from_url(list(server = "https://foo.com"), "http://foo.com/bar/foo-com_baz")
[1] "_baz"
```